### PR TITLE
ci: add publish workflow

### LINKED ISSUE #110: Implement packaging and CD
In fact, users prefer to use a package managers like `yum`, `apt` ... to get the tool they need instead of `git clone ... && mage build`. We need to implement packaging to `rpm` and `deb` packages and add GHA to deliver packages to our S3 repository ([modules](https://rws.tarantool.org/release/modules)).

### PR DESCRIPTION
**Pull request by 0x501D**
Added CI workflow to build and publish RPM and DEB packages to RWS.
    
Closes #110

### CODE REVIEW

File: .gen_tarballs.sh [hunk 1]
```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
+set -u
+
+# This script generates tarballs nedded to create the Gentoo ebuilds.
```
Comment (DifferentialOrange) [inline .gen_tarballs.sh:4]:
```suggestion
# This script generates tarballs needed to create the Gentoo ebuilds.
```

File: .gen_tarballs.sh [hunk 2]
```diff
@@ -0,0 +1,96 @@
+#!/bin/sh
```
Comment (DifferentialOrange) [inline .gen_tarballs.sh:1]: The file is not executable.

File: magefile.go [hunk 3]
```diff
@@ -153,6 +153,7 @@ func Build() error {
 	err := sh.RunWith(
 		getBuildEnvironment(), goExecutableName, "build",
 		"-o", ttExecutableName,
+		"-tags=dev go_tarantool_ssl_disable",
```
Comment (DifferentialOrange) [inline magefile.go:156]: What does `dev` build tag do?

File: .github/workflows/publish.yml [hunk 4]
```diff
@@ -0,0 +1,85 @@
+name: publish
```
Comment (DifferentialOrange) [inline .github/workflows/publish.yml:1]: It's more like `cd: add publish workflow `

**Review by DifferentialOrange — COMMENTED** [review]
Thank you for the PR! I have left several minor comments.

File: magefile.go (hunk 3 above)
Comment (0x501D) [reply magefile.go:156]: Disables linking with OpenSSL. We do not need this in TT. Also this needed for static build.

File: .github/workflows/publish.yml (hunk 4 above)
Comment (0x501D) [reply .github/workflows/publish.yml:1]: Fixed

File: .github/workflows/publish.yml [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+
+on:
+  push:
+    branches: [master]
+    tags: ['*']
+
+env:
+  GO_VERSION: 1.18
```
[the remark this answers is not in the document]
Comment (0x501D) [reply .github/workflows/publish.yml:9]: Fixed

File: .gen_tarballs.sh (hunk 2 above)
Comment (0x501D) [reply .gen_tarballs.sh]: Fixed

File: .gen_tarballs.sh (hunk 1 above)
Comment (0x501D) [reply .gen_tarballs.sh]: Fixed

File: magefile.publish.go [hunk 6]
```diff
@@ -0,0 +1,137 @@
...
+var targetDistros = []Distro{
+	{OS: "el", Dist: "7"},
+
+	{OS: "fedora", Dist: "34"},
+	{OS: "fedora", Dist: "35"},
+	{OS: "fedora", Dist: "36"},
+
+	{OS: "ubuntu", Dist: "xenial"}, // 16.04
+	{OS: "ubuntu", Dist: "bionic"}, // 18.04
+	{OS: "ubuntu", Dist: "focal"},  // 20.04
```
[the remark this answers is not in the document]
Comment (0x501D) [reply magefile.publish.go:31]: Included same versions as Tarantool uses in CI.

File: magefile.go (hunk 3 above)
Comment (0x501D) [reply magefile.go]: Fixed

File: magefile.publish.go [hunk 7]
```diff
@@ -0,0 +1,137 @@
+//go:build mage
+// +build mage
```
Comment (LeonidVas) [inline magefile.publish.go:2]: `go:build` [is enough](https://stackoverflow.com/a/68361186)

File: magefile.publish.go [hunk 8]
```diff
@@ -0,0 +1,137 @@
...
+	}
+
+	if distro.OS == "ubuntu" || distro.OS == "debian" {
+		return []string{"*.deb", "*.dsc"}, nil
+	}
+
+	return nil, fmt.Errorf("Unknown OS: %s", distro.OS)
+}
+
+func PublishRWS() error {
```
Comment (LeonidVas) [inline magefile.publish.go:85]: Description?

File: tarantool.yaml.default [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+    data_dir: /var/lib/tarantool
+
+    # Directory that stores binary files.
+    bin_dir: bin
+
+    # Directory that stores Tarantool header files.
+    inc_dir: include
+
+    # Path to directory that stores all applications.
+    instances_enabled: /etc/tarantool/instances.available
```
Comment (LeonidVas) [inline tarantool.yaml.default:36]: `instances.available` -> `instances.enabled`

File: tarantool.yaml.default [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+    # Directory where all the instance artifacts ar stored.
+    data_dir: /var/lib/tarantool
+
+    # Directory that stores binary files.
+    bin_dir: bin
+
+    # Directory that stores Tarantool header files.
+    inc_dir: include
+
+    # Path to directory that stores all applications.
```
Comment (LeonidVas) [inline tarantool.yaml.default:35]: In fact, these will be repositories of links to applications. I think this should be reflected in the comment.

File: tarantool.yaml.default [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+    # Path to file with credentials for downloading Tarantool Enterprise Edition.
+    # credential_path: /path/to/file 
+
+  templates:
+    # The path to templates search directory.
+    - path: templates
+
+  repo:
+    # Directory that stores rocks files.
+    rocks: ""
```
Comment (LeonidVas) [inline tarantool.yaml.default:48]: Not yet implemented, better to remove from the default config.

File: .goreleaser.yml [hunk 12]
```diff
@@ -0,0 +1,83 @@
...
+
+nfpms:
+  -
+    id: "tt"
+    builds: ['tt']
+    formats:
+        - deb
+        - rpm
+    homepage: "https://github.com/tarantool/tt"
+    maintainer: "Pavel Balaev <balaev@tarantool.org>"
```
Comment (LeonidVas) [inline .goreleaser.yml:63]: https://github.com/tarantool/ecosystem-team#gatekeepers (yet Leonid Vasiliev :) )

**Review by LeonidVas — CHANGES_REQUESTED** [review]
Hi! Thank you for the patchset.
See several comments bellow.

File: .gen_tarballs.sh [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+done
+
+if [ $# == 0 ]; then
+    die "Usage: ${0} -t <tag>"
+fi
+
+TMPDIR="/tmp/gentoo_tarballs"
+TT_DIR="${TMPDIR}/tt-${TAG}"
+
+# cleanup
```
[the remark this answers is not in the document]
Comment (0x501D) [reply .gen_tarballs.sh]: FIxed

File: magefile.publish.go (hunk 7 above)
Comment (0x501D) [reply magefile.publish.go]: Fixed

File: magefile.publish.go (hunk 8 above)
Comment (0x501D) [reply magefile.publish.go:85]: Added.

File: tarantool.yaml.default (hunk 9 above)
Comment (0x501D) [reply tarantool.yaml.default]: Fixed

File: tarantool.yaml.default (hunk 10 above)
Comment (0x501D) [reply tarantool.yaml.default:35]: Comment added.

File: tarantool.yaml.default (hunk 11 above)
Comment (0x501D) [reply tarantool.yaml.default]: Removed.

File: .goreleaser.yml (hunk 12 above)
Comment (0x501D) [reply .goreleaser.yml]: Fixed. AFAIK this is a package maintainer record.

**Review by LeonidVas — APPROVED** [review]

**Pull request merged**